### PR TITLE
Reuse pred buf to reduce memory

### PR DESCRIPTION
**Pull request by xrayleigh2000**
Signed-off-by: xrayleigh2000 <xrayleigh2000@gmail.com>

### CODE REVIEW

**Comment by adamjw24** [comment]
Thank you for looking into this.

Did you have a look at the performance? Especially the INTER task is great for parallelization, and we used to rely very much on it (we didn't reevaluate the architecture in quite some time tho, and a lot changed since).

Overall this would save around 60MB for UHD decoding indepdent of threading, right? (two full frames of `Pel`)

Because of some upcoming dates (CES/vacation) we will only be able to properly have a look at this towards the end of January, so please be patient. If you could in the meantime provide some additional data like memory savings, performance impact, it would be highly appreciated.

Also, the CI seems broken, please have a look.

File: source/Lib/DecoderLib/DecCu.cpp [hunk 1]
```diff
@@ -910,4 +951,73 @@ void DecCu::xDeriveCUMV( CodingUnit &cu, MotionHist& hist )
...
+  }
+
+  return ret;
+}
+
+const CPelUnitBuf DecCu::getPredBuf(const CodingUnit &unit) const
+{
+  CPelUnitBuf ret;
+  ret.chromaFormat = unit.chromaFormat;
+  ret.bufs.resize( 3 );
```
Comment (adamjw24) [inline source/Lib/DecoderLib/DecCu.cpp:994]: Consider using `resize_noinit` with proper number of components, similar to the non-const variant. All the stuff will be manually initalized after.

**Comment by K-os** [comment]
Hi @xrayleigh2000, thanks for the PR.

I had a quick look into it and tested it on my machine. As @adamjw24 expected, it is much slower than the original implementation, because it significantly reduces parallelism. On my machine running with 10 to 20 threads decoding FullHD content was 1.3 to 1.4 times slower.

**Comment by xrayleigh2000** [comment]
Hi @adamjw24 @K-os, Thanks for your reply.
Under the current architecture, pred buf is difficult to modify to CU-level size. I am going to close this pull request. For subsequent optimization, I will submit new pull requests and provide more test data.

**Pull request closed without merging**